client/my-sites/stats/activity-log-item/index.jsx

```diff
@@ -45,113 +47,97 @@ class ActivityLogItem extends Component {
 				'user',
 				'widget',
 			] ).isRequired,
-			name: PropTypes.string.isRequired,
-			ts_utc: PropTypes.number.isRequired,
-
-			actor: PropTypes.shape( {
-				display_name: PropTypes.string,
-				login: PropTypes.string,
-				translated_role: PropTypes.string,
-				user_email: PropTypes.string,
-				user_roles: PropTypes.string,
-				wpcom_user_id: PropTypes.number,
-				avatar_url: PropTypes.string,
-			} ),
-
-			object: PropTypes.shape( {
-				attachment: PropTypes.shape( {
-					mime_type: PropTypes.string,
-					id: PropTypes.number.isRequired,
-					title: PropTypes.string.isRequired,
-					url: PropTypes.shape( {
-						host: PropTypes.string.isRequired,
-						url: PropTypes.string.isRequired,
-						host_reversed: PropTypes.string.isRequired,
-					} ).isRequired,
-				} ),
-
-				comment: PropTypes.shape( {
-					approved: PropTypes.bool.isRequired,
-					id: PropTypes.number.isRequired,
-				} ),
-
-				core: PropTypes.shape( {
-					new_version: PropTypes.string,
-					old_version: PropTypes.string,
-				} ),
-
-				error: PropTypes.shape( {
-					error_code: PropTypes.string,
-					error_message: PropTypes.string,
-				} ),
-
-				menu: PropTypes.shape( {
-					id: PropTypes.number,
-					name: PropTypes.string,
-				} ),
-
-				plugin: PropTypes.oneOfType( [
-					PropTypes.shape( {
-						name: PropTypes.string,
-						previous_version: PropTypes.string,
-						slug: PropTypes.string,
-						version: PropTypes.string,
-					} ),
-					PropTypes.arrayOf(
-						PropTypes.shape( {
-							name: PropTypes.string,
-							previous_version: PropTypes.string,
-							slug: PropTypes.string,
-							version: PropTypes.string,
-						} ),
-					),
-				] ),
-
-				post: PropTypes.shape( {
-					id: PropTypes.number.isRequired,
-					status: PropTypes.string.isRequired,
-					type: PropTypes.string,
-					title: PropTypes.string,
-				} ),
-
-				term: PropTypes.shape( {
-					id: PropTypes.number.isRequired,
-					title: PropTypes.string.isRequired,
-					type: PropTypes.string.isRequired,
-				} ),
-
-				theme: PropTypes.oneOfType( [
-					PropTypes.arrayOf(
-						PropTypes.shape( {
-							name: PropTypes.string,
-							slug: PropTypes.string,
-							uri: PropTypes.string,
-							version: PropTypes.string,
-						} )
-					),
-					PropTypes.shape( {
-						name: PropTypes.string,
-						slug: PropTypes.string,
-						uri: PropTypes.string,
-						version: PropTypes.string,
-					} ),
-				] ),
-
-				user: PropTypes.shape( {
-					display_name: PropTypes.string,
-					external_user_id: PropTypes.string,
-					login: PropTypes.string,
-					wpcom_user_id: PropTypes.number,
-				} ),
-
-				widget: PropTypes.shape( {
-					id: PropTypes.number,
-					name: PropTypes.string,
-					sidebar: PropTypes.string,
-				} ),
-			} ),
+			activityIcon: PropTypes.string.isRequired,
+			activityId: PropTypes.string.isRequired,
+			activityName: PropTypes.oneOf( [
```

Comment:
I do have reservations about the usefulness and maintainability of these huge prop types, especially given that we have the schemas as well. This is only an observation, though, carry on with them if you think they add value.

Reply:
I agree. I've moved the complete, strict details we have to the schema in bca217d and relaxed the propTypes in 477374c